src/libbson/src/bson/bson-cmp.h

```diff
@@ -118,77 +118,68 @@ BSON_CMP_SET (greater_equal,
+   {                                                                       \
+      return bson_cmp_greater_equal_su (value, 0u) &&                      \
+             bson_cmp_less_equal_su (value, max);                          \
+   }                                                                       \
+                                                                           \
+   static BSON_INLINE bool BSON_CONCAT3 (bson_in_range, Type, _unsigned) ( \
+      uint64_t value)                                                      \
+   {                                                                       \
+      return bson_cmp_less_equal_uu (value, max);                          \
    }
 
-BSON_IN_RANGE_SET_SIGNED (signed_char, SCHAR_MIN, SCHAR_MAX)
-BSON_IN_RANGE_SET_SIGNED (short, SHRT_MIN, SHRT_MAX)
-BSON_IN_RANGE_SET_SIGNED (int, INT_MIN, INT_MAX)
-BSON_IN_RANGE_SET_SIGNED (long, LONG_MIN, LONG_MAX)
-BSON_IN_RANGE_SET_SIGNED (long_long, LLONG_MIN, LLONG_MAX)
+BSON_IN_RANGE_SET_SIGNED (_signed_char, SCHAR_MIN, SCHAR_MAX)
+BSON_IN_RANGE_SET_SIGNED (_short, SHRT_MIN, SHRT_MAX)
+BSON_IN_RANGE_SET_SIGNED (_int, INT_MIN, INT_MAX)
+BSON_IN_RANGE_SET_SIGNED (_long, LONG_MIN, LONG_MAX)
+BSON_IN_RANGE_SET_SIGNED (_long_long, LLONG_MIN, LLONG_MAX)
 
-BSON_IN_RANGE_SET_UNSIGNED (unsigned_char, UCHAR_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (unsigned_short, USHRT_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (unsigned_int, UINT_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (unsigned_long, ULONG_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (unsigned_long_long, ULLONG_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_unsigned_char, UCHAR_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_unsigned_short, USHRT_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_unsigned_int, UINT_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_unsigned_long, ULONG_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_unsigned_long_long, ULLONG_MAX)
 
-BSON_IN_RANGE_SET_SIGNED (int8_t, INT8_MIN, INT8_MAX)
-BSON_IN_RANGE_SET_SIGNED (int16_t, INT16_MIN, INT16_MAX)
-BSON_IN_RANGE_SET_SIGNED (int32_t, INT32_MIN, INT32_MAX)
-BSON_IN_RANGE_SET_SIGNED (int64_t, INT64_MIN, INT64_MAX)
+BSON_IN_RANGE_SET_SIGNED (_int8, INT8_MIN, INT8_MAX)
+BSON_IN_RANGE_SET_SIGNED (_int16, INT16_MIN, INT16_MAX)
+BSON_IN_RANGE_SET_SIGNED (_int32, INT32_MIN, INT32_MAX)
+BSON_IN_RANGE_SET_SIGNED (_int64, INT64_MIN, INT64_MAX)
 
-BSON_IN_RANGE_SET_UNSIGNED (uint8_t, UINT8_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (uint16_t, UINT16_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (uint32_t, UINT32_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (uint64_t, UINT64_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_uint8, UINT8_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_uint16, UINT16_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_uint32, UINT32_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_uint64, UINT64_MAX)
 
-BSON_IN_RANGE_SET_SIGNED (ssize_t, SSIZE_MIN, SSIZE_MAX)
-BSON_IN_RANGE_SET_UNSIGNED (size_t, SIZE_MAX)
+BSON_IN_RANGE_SET_SIGNED (_ssize, SSIZE_MIN, SSIZE_MAX)
+BSON_IN_RANGE_SET_UNSIGNED (_size, SIZE_MAX)
 
 #undef BSON_IN_RANGE_SET_SIGNED
 #undef BSON_IN_RANGE_SET_UNSIGNED
 
 
```

Comment:
Can these macros be added back to avoid breaking API.

Consider prepending an `_`.

```suggestion
#define bson_in_range_signed(Type, value) \
   BSON_CONCAT3 (bson_in_range, _##Type, _signed) (value)
```

Reply:
I am surprised this prevents the undesirable type macro expansion behavior. I had incorrectly assumed `Type`  would still be expanded before the `_` prefix is concatenated. I will gladly apply this solution over removal of these macros.

Reply:
Would adding an underscore prefix _and_ suffix to the type and removing the prefix on `_signed` and `_unsigned` be preferable, or is there no risk that we'll run into a case where `_size_t` is a definition?

Just asking because I think I have seen leading or trailing underscores appear in types before (usually within internal headers).

Reply:
Actually, the `_##Type` solution should be applicable to the `BSON_IN_RANGE_SET_(UN)SIGNED` macros as well, which would obviate the need for the underscore handling as an expansion workaround, which further reduces the scope of changes necessary.